protocol/consensus/hotstuff.go

```diff
@@ -39,31 +41,36 @@ func NewHotStuff(
 			auth,
 			states,
 		),
+		states:         states,
 		leaderRotation: leaderRotation,
 		sender:         sender,
 	}
 }
 
-func (cs *HotStuff) SendPropose(proposal *hotstuff.ProposeMsg, pc hotstuff.PartialCert) {
-	cs.votingMachine.CollectVote(hotstuff.VoteMsg{ID: cs.config.ID(), PartialCert: pc})
-	cs.sender.Propose(proposal)
+func (hs *HotStuff) Disseminate(proposal *hotstuff.ProposeMsg, pc hotstuff.PartialCert) {
+	hs.votingMachine.CollectVote(hotstuff.VoteMsg{
+		ID:          hs.config.ID(),
+		PartialCert: pc,
+	})
+	hs.sender.Propose(proposal)
 }
 
-// SendVote disseminates or stores a valid vote depending on replica being voter or leader in the next view.
-func (cs *HotStuff) SendVote(proposal *hotstuff.ProposeMsg, pc hotstuff.PartialCert) {
-	leaderID := cs.leaderRotation.GetLeader(proposal.Block.View() + 1)
-	if leaderID == cs.config.ID() {
+// Aggregate disseminates or stores a valid vote depending on replica being voter or leader in the next view.
```

Comment:
Is this doc still accurate? At least it is a bit confusing that aggregate does dissemination.